src/backend/joanie/core/api.py

```diff
@@ -112,10 +126,12 @@ def get_queryset(self):
         )
 
     def perform_create(self, serializer):
-        """Force the order's "owner" field to the logged-in user."""
+        """Force the order's "owner" field to the logged-in user and proceed to payment"""
         username = self.request.user.username
         owner = models.User.objects.get_or_create(username=username)[0]
-        serializer.save(owner=owner)
+        order = serializer.save(owner=owner)
+        if order.price and self.request.data.get("credit_card"):
```

Comment:
I'm thinking maybe these 2 checks should be done inside the `proceed_to_payment` method so that any call to this function can benefit from them:
- what do we do if the price is null or 0: mark the order as paid without looking for a credit card
- what do we do if the price is not null but no credit card is passed: use the default credit card? and if there's no default credit card raise a ValidationError? => in this case your API must return it as a 400 (add a test for this)